ansible_bender/core.py

```diff
@@ -389,3 +395,206 @@ def get_build_and_metadata(self):
+
+                    # key can be prepended with ansible_bender_ or ansible_bender
+                    # if it is prepended with ansible_bender_, remove the _
+                if new_key.startswith("_"):
+                    new_key = new_key[1:]
+
+                    # If the new_key is blank
+                    # then the "ansible_bender" variable was used
+                    # Add the variables to the ansible_variables
+                if new_key == "":
+                    ansible_variables.update(value)
+                else:
+                    ansible_variables[new_key] = value
+
+        return ansible_variables
+
+    def _get_vars_from_playbook(self, host_name) -> dict:
+        """
+        Parse the playbook variables related to the ansible_bender information
+        """
+        playbook_variables = {}
+
+        try:
+            for host in self._hosts_in_playbook:
+                if host.get("hosts") == host_name:
+                    playbook_variables = self._get_variables(playbook_variables, host.get("vars"))
+        except Exception as e:
+            print("Error getting playbook variables for host: " + host_name + " - " + str(e))
+            pass
+
+        return playbook_variables
+
+    def _get_vars_for_host(self, host_name) -> dict:
+        """
+        Parse the host variables from the inventory information
+        """
+        ansible_variables = {}
+        try:
+            ansible_host = self._inventory.get_host(host_name)
+            all_vars_for_host = self._variable_manager.get_vars(host=ansible_host, 
+                                                            include_hostvars=True, 
+                                                            include_delegate_to=True)
+
+            # find prepended variables
+            self._get_variables(ansible_variables, all_vars_for_host)                
+        except Exception as e:
+            print("Error getting host and group vars for host: " + host_name + " - " + str(e))
+            pass
+
+        # Append the playbook variables to the ansible variables
+        ansible_variables.update(self._get_vars_from_playbook(host_name))
+
+        return ansible_variables
+
+    
+    def get_build_and_metadata(self):
+        """
+        Find all hosts in the playbook and get their variables
+        return the build and imagemetadata information
+        """
+        if self._inventory_path:
+            # Ansible API objects used to pull variables
+            relative_playbook_file_path = os.path.basename(self._playbook_path)
+            self._loader = DataLoader()
+            self._loader.set_basedir(self._base_dir)
+            self._inventory = InventoryManager(loader=self._loader, sources=self._inventory_path)
+            self._variable_manager = VariableManager(loader=self._loader, inventory=self._inventory)
+            self._hosts_in_playbook = self._loader.load_from_file(relative_playbook_file_path)
```

Comment:
could you please define these in `__init__`? so that it's easy to see what class properties are available just by looking at `__init__`

Reply:
@TomasTomecek - how about as private class variables at the top of the class? I'll make the change and you let me know how it looks. I agree easily understanding the private class properties is a good thing. Technically though, if there is no inventory file they don't need be initialized. I check if the inventory is found in `AnsibleVarsParser.get_build_and_metdata()`. I could also check in `__init__`

Reply:
perfect, thank you for the change, that's what I had in mind